src/utilities.jl

```diff
@@ -0,0 +1,89 @@
+using MCMCChains, StatsFuns
+export pointwise_loglikes, compute_loo
+
+"""
+    pointwise_loglikes(chain::Chains, data, ll_fun)
+
+Computes the pointwise log likelihoods where [d,s,c] corresponds to log likelihood of 
+evaluated at datapoint d, sample s, for chain c. 
+
+- `chain`: a chain object from MCMCChains
+- `data`: a vector of data used to estimate parameters of the model 
+- `ll_fun`: a function that computes the log likelihood of a single data point: f(θ1, ..., θn, data), where θi is the ith parameter
+"""
+function pointwise_loglikes(chain::Chains, data, ll_fun)
+    samples = Array(Chains(chain, :parameters).value)
+    pointwise_loglikes(samples, data, ll_fun)
+end
+
+"""
+    pointwise_loglikes(samples::Array{Float64,3}, data, ll_fun)
```

Comment:
Should not be limited to `Float64` -- the only thing we can say is that it will be of some type `{F <: AbstractFloat}`. Float32s and Float16s are common in larger models to speed up computing.